lib/Parse/ParseDecl.cpp

```diff
@@ -6748,6 +6748,18 @@ Parser::parseDeclVar(ParseDeclOptions Flags,
     // If we syntactically match the second decl-var production, with a
     // var-get-set clause, parse the var-get-set clause.
     if (Tok.is(tok::l_brace)) {
+
+      // Skip parsing the var-get-set clause if '{' is at start of line
+      // and next token is not 'didSet' or 'willSet'. Parsing as 'do'
+      // statement gives useful errors for missing 'do' before brace.
+      // See SR-14836.
+      Token NextToken = peekToken();
+      if ((PatternInit != nullptr) && Tok.isAtStartOfLine() &&
+          !NextToken.isContextualKeyword("didSet") &&
+          !NextToken.isContextualKeyword("willSet")) {
+        return makeResult(Status);
```

Comment:
I noticed `didSet`/`willSet` accepts some attributes. for example, this is accepted.
```swift
var value: Int = 1
{
    @available(*, unavailable)
    didSet { print("didSet") }
}
```
Although this seems to be not working correctly, we can't break this behavior.

Could you call `isStartOfGetSetAccessor()` on `ParseExpr.cpp` instead of this manual `NextToken.isContextualKeyword("didSet")` checking? That function skips attributes correctly. You can make `isStartOfGetSetAccessor()` be a `Parser` method.